curriculum/challenges/english/25-front-end-development/lab-leap-year-calculator/66c06fad3475cd92421b9ac2.md

```diff
@@ -0,0 +1,67 @@
+---
+id: 66c06fad3475cd92421b9ac2
+title: Build a Leap Year Calculator
+challengeType: 14
+dashedName: build-a-leap-year-calculator
+---
+
+# --description--
+
+A leap year is a year that is divisible by 4, except for years that are divisible by `100` and not divisible by `400`. For example, `2000` is a leap year, but 1900 is not. Also, a leap year has an extra day in February, which is the 29th day of the month.
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. Declare a variable `year` that stores the value of the year to check.
+
+2. Use an `if/ else` statement or a ternary operator to check if the year is a leap year. 
+
+3. To check if the year is a leap year, fulfill the following conditions:
+   
+- If the year is divisible by `4`, then it is a leap year.
+- If the year is divisible by `100`, then it is not a leap year.
+- If the year is divisible by `400`, then it is a leap year.
+
+1. If the year is a leap year, output `[year] is a leap year.` to the console. Otherwise, output `[year] is not a leap year.`. You will replace [year] with the year variable.
+
+# --hints--
+
+You should declare a variable `year` and assign it a value to check if it is a leap year.
+
+```js
+assert.isDefined(year);
+```
+
+With `2024` as the value of the `year` variable, the output should be `2024 is a leap year.`.
```

Comment:
@jdwilkin4 
Is there a way to get what is being printed to the console?
Or is there a way where we can run the user entered code and then match the results with a value?

Reply:
As of right now, no. Chai doesn't have a helpful function for that.

The easiest way is to have campers assign the result to the variable and then have them log the variable.
That still gives them the flexibility of using an if/else or ternary to output the correct string whether it is a leap year or not.